### PR TITLE
I added query APIs

### PR DESCRIPTION
**Pull request by dairyo**
***This PR is not buildable but this PR is not merged to develop. Please feel free to merge this.***

I added following query APIs:

  * query without time range
  * query with time range
  * count
  * aggregate

I will write API document after we agree this API.

Related PR: #265

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/ThingIFAPI.swift [hunk 1]
```diff
@@ -736,6 +736,73 @@ open class ThingIFAPI: NSObject, NSCoding {
...
+    {
+        // TODO: implement me.
+    }
+
+    open func count(
+      _ alias: String,
+      _ range: (from: Date, to: Date),
+      _ field: (name: String, type: CountFieldType),
+      clause: QueryClause? = nil,
+      firmeware: String? = nil,
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/ThingIFAPI.swift]: Better to use `firmwareVersion`

File: ThingIFSDK/ThingIFSDK/ThingIFAPI.swift [hunk 2]
```diff
@@ -736,6 +736,73 @@ open class ThingIFAPI: NSObject, NSCoding {
...
+        // TODO: implement me.
+    }
+
+    open func aggregate<ReturnedValue>(
+      _ alias: String,
+      _ range: (from: Date, to: Date),
+      _ function: Aggregation.FunctionType,
+      _ field: (name: String, type: Aggregation.FieldType),
+      clause: QueryClause? = nil,
+      firmeware: String? = nil,
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/ThingIFAPI.swift]: Better to name `firmwareVersion`

File: ThingIFSDK/ThingIFSDK/ThingIFAPI.swift [hunk 3]
```diff
@@ -736,6 +736,73 @@ open class ThingIFAPI: NSObject, NSCoding {
...
+      _ range: (from: Date, to: Date),
+      _ field: (name: String, type: CountFieldType),
+      clause: QueryClause? = nil,
+      firmeware: String? = nil,
+      completionHandler: @escaping (
+        (result:
+           (count: Int,
+            range: (from: Date, to: Date),
+            queryDescription: String
+           )?,
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/ThingIFAPI.swift]: result is array

File: ThingIFSDK/ThingIFSDK/ThingIFAPI.swift [hunk 4]
```diff
@@ -736,6 +736,73 @@ open class ThingIFAPI: NSObject, NSCoding {
...
+      clause: QueryClause? = nil,
+      firmeware: String? = nil,
+      completionHandler: @escaping(
+        (result:
+           (
+             value: ReturnedValue,
+             range: (from: Date, to: Date),
+             objects: [[String : Any]],
+             queryDescription: String
+           )?,
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/ThingIFAPI.swift]: result is array

**Review by yongpingchen — COMMENTED** [review]
BTW, can you add snippets on PR description?

**Comment by dairyo** [comment]
@yongpingchen 

I fixed according to your comment.

> BTW, can you add snippets on PR description?

OK. I'm going to write snippets for all APIs to each PRs👍

**Pull request merged**